### PR TITLE
docs(provider): clarify opencode is cloud-only, recommend ollama for custom endpoints

### LINKED ISSUE #582: provider opencode routes to OpenCode.ai directly, ignores base_url override — ollama provider needed for custom endpoints
## Problem

When using `--provider opencode`, WUPHF routes all LLM calls directly to `https://opencode.ai/zen/go/v1/chat/completions`, ignoring any `provider_endpoints.opencode.base_url` set in `~/.wuphf-spaces/main/.wuphf/config.json`.

This means users who want to route through a local OpenAI-compatible gateway (e.g. OmniRoute, LiteLLM, or any `/v1/chat/completions` proxy) **cannot use the `opencode` provider** for that purpose.

### Error observed

```
{"type":"error","error":{"type":"CreditsError","message":"Insufficient balance. Manage your billing here: https://opencode.ai/workspace/wrk_01KNHSCXE80MC56PSGBHDJ4XW7/billing"}}
```

Metadata shows `"url":"https://opencode.ai/zen/go/v1/chat/completions"` — hardcoded, not our configured endpoint.

## Expected behavior

One of:
1. `opencode` provider respects `base_url` in `provider_endpoints` config (or `WUPHF_OPENCODE_BASE_URL` env var), OR
2. Documentation clearly states that `opencode` is a cloud-only provider with no base_url override, and recommends `ollama` for custom OpenAI-compatible endpoints

## Workaround found

Using `--provider ollama` with `WUPHF_OLLAMA_BASE_URL` env var works correctly — the ollama provider respects the custom base URL and routes through the local gateway as expected.

```
WUPHF_OLLAMA_BASE_URL="http://127.0.0.1:20128/v1" \
WUPHF_OLLAMA_MODEL="openai/gpt-5.4-mini" \
wuphf --provider ollama --memory-backend none --no-open
```

## Additional issues found during setup

1. **`WUPHF_NO_NEX` env var has no effect.** The binary still prompts `Connect Nex for memory and context? [Y/n]` on stdin. The string `WUPHF_NO_NEX` appears in the JS frontend bundle but is not honored by the Go binary. Must run in PTY mode and manually answer `n`.

2. **Onboarding doesn't pick up `--memory-backend` CLI flag or config.json.** Config file has `"memory_backend": "none"` and CLI passes `--memory-backend none`, but the Settings UI shows `Nex` as the memory backend. The onboarding flow seems to default to `nex` regardless. The Settings save via UI works, but requires a shred+relaunch cycle to actually take effect.

3. **`--provider` flag doesn't override agent harness type.** After re-onboarding with `--provider ollama`, the agents in the sidebar still show "Claude Code harness". The provider dropdown in Settings shows the correct value, but agents created during onboarding retain their original harness. The UI warning says to `wuphf shred` and relaunch, but the shred button in Danger Zone didn't trigger (possible rendering issue with the confirmation modal).

## Environment

- WUPHF binary: `wuphf-0.99.7` (cached Go binary)
- npm wrapper: `wuphf@0.97.0`
- OS: Linux (Arch/rolling)
- Shell: bash with PTY

### PR DESCRIPTION
**Pull request by mvanhorn**
## Summary
Documents that `--provider opencode` shells out to the `opencode` CLI binary and that `provider_endpoints.opencode.base_url` is not consulted on that path. Recommends `--provider ollama` with `WUPHF_OLLAMA_BASE_URL` for users who want a custom OpenAI-compatible endpoint. Also fixes a `TestSeedCompanyContext/pdftotext_not_found` subtest that fails on macOS dev boxes with `poppler` installed.

## Why this matters
- Issue: [#582](https://github.com/nex-crm/wuphf/issues/582) - "provider opencode routes to OpenCode.ai directly, ignores base_url override". The reporter explicitly offers two acceptable resolutions: (1) make opencode respect `provider_endpoints.opencode.base_url`, or (2) clarify in docs that opencode is a CLI-subprocess provider with no base_url override and recommend ollama for custom endpoints. This PR ships option 2.
- `internal/provider/opencode.go` registers `KindOpencode` with `CreateOpencodeCLIStreamFn`, which shells out to the `opencode` binary. WUPHF makes no HTTP requests to opencode.ai - the `https://opencode.ai/zen/...` URL the reporter saw is emitted by the opencode binary itself. Option 1 would require the opencode CLI to support endpoint override, which is upstream-unknown.
- The reporter already confirmed in the issue that `--provider ollama` with `WUPHF_OLLAMA_BASE_URL` works for custom OpenAI-compatible endpoints (LiteLLM, OmniRoute, local proxies). The docs change makes this discoverable.

## Changes
- `README.md` adds an `Opencode and custom endpoints` subsection in the providers area. Explains the CLI-subprocess shape and points users at `--provider ollama` for custom endpoints, with the reporter's working snippet as a code block.
- `internal/operations/company_seed_test.go` - adds an optional `setup func(t *testing.T)` field to `TestSeedCompanyContext`, used only by the `pdftotext not found` case to `t.Setenv("PATH", "")`. The case relies on `extractPDF`'s `exec.LookPath("pdftotext")` returning an error, but on a Mac with `poppler` installed `pdftotext` IS on PATH and the not-installed warning never fires. Scrubbing PATH in the per-case setup makes the test deterministic on any host.

## Testing
- `bash scripts/test-go.sh ./internal/operations` -> ok, including the previously flaky `pdftotext_not_found` subtest.
- Full pre-push hook suite green: build, complexity-baseline, file-size, smoke, vhs, web-unit (1438 tests), go-unit.

Fixes #582

AI was used for assistance.

## Summary by CodeRabbit

* **Documentation**
  * Added guidance for configuring custom endpoints, including support for the new provider option and integration with Ollama for OpenAI‑compatible services.
* **Tests**
  * Improved test coverage for document-processing edge cases to make missing-tool scenarios deterministic.

### CODE REVIEW

**Comment by FranDias** [comment]
👀

File: README.md [hunk 1]
```diff
@@ -104,6 +104,22 @@ if I say yes. If I am not logged in, just open https://wuphf.team.
...
+consulted.
+
+For custom OpenAI-compatible endpoints such as LiteLLM, OmniRoute, or local
+proxies, use `--provider ollama` and set `WUPHF_OLLAMA_BASE_URL` or
+`provider_endpoints.ollama.base_url`:
+
+```bash
+WUPHF_OLLAMA_BASE_URL="http://127.0.0.1:20128/v1" \
+WUPHF_OLLAMA_MODEL="openai/gpt-5.4-mini" \
+wuphf --provider ollama --memory-backend none --no-open
```
[the remark this answers is not in the document]
Comment (mvanhorn) [reply README.md:120]: Already in the options table at line 96. The `--provider <name>` row lists `claude-code, codex, opencode, ollama`. No contradiction here; resolving.

**Review by FranDias — APPROVED** [review]
Docs change + a deterministic test fix. CI green after merging main. LGTM.
